apps/els_lsp/src/els_references_provider.erl

```diff
@@ -69,14 +72,23 @@ find_references(Uri, #{ kind := Kind
   {F, A, _Index} = Id,
   Key = {els_uri:module(Uri), F, A},
   find_references_for_id(Kind, Key);
-find_references(_Uri, #{kind := Kind, id := Name})
-  when Kind =:= record_expr;
-       Kind =:= record ->
-  find_references_for_id(Kind, Name);
-find_references(_Uri, #{kind := Kind, id := Name})
-  when Kind =:= macro;
+find_references(Uri, Poi = #{kind := Kind})
+  when Kind =:= record;
+       Kind =:= record_def_field;
        Kind =:= define ->
-  find_references_for_id(Kind, Name);
+  uri_pois_to_locations(
+    find_scoped_references_for_def(Uri, Poi));
+find_references(Uri, Poi = #{kind := Kind})
+  when Kind =:= record_expr;
+       Kind =:= record_field;
+       Kind =:= macro ->
+  case els_code_navigation:goto_definition(Uri, Poi) of
+    {ok, DefUri, DefPoi} ->
+      uri_pois_to_locations(
+        find_scoped_references_for_def(DefUri, DefPoi));
+    {error, _} ->
+      []
```

Comment:
FIXME: could still look for references in the current document, even if definition is not yet declared